### PR TITLE
Allow Helm parameters to force ambiguously-typed values to be strings

### PR DESCRIPTION
**Pull request by jutley**
This adds support for setting Helm parameters as strings when they would otherwise be interpreted as numbers or booleans. This addresses #1846.

Currently, I have not added any way to do this from the CLI or web UI, only from Application manifests. I can look into making these changes as well, though I would appreciate suggestions on how the interface should look. Should the CLI have a separate flag (as Helm does) or add some indication to the current flag (such as with the `component` portion of Ksonnets `component=key=value` parameters. 

Also, Ksonnet has this same issue which I also haven't addressed. The way ksonnet parameters are sanitized makes this more complicated than with Helm, so I think it deserves to be addressed in a separate PR.

### CODE REVIEW

**Comment by jutley** [comment]
Struggling with making CI happy. I'll clean up the commits before anything is merged.

**Comment by jutley** [comment]
On my laptop, both the unit tests and e2e tests pass. I do not know why they are failing on CircleCI. They used to pass, but then the build would fail at `Check nothing has changed`. I ran the `make pre-commit` command as recommended at https://argoproj.github.io/argo-cd/developer-guide/ci/, and now the tests won't pass. I'm very confused. Any pointers?

File: manifests/base/kustomization.yaml
Comment (alexmt) [inline manifests/base/kustomization.yaml]: Thank you for working on this feature @jutley! CI is not happy because of this change. We are still using Kustomize v2.0.0 . Can you please revert this change?

Comment (jutley) [reply manifests/base/kustomization.yaml]: Okay, I could see this being the issue! I'll get that installed and try again!

I'll also point out that I would prefer to just download the patch artifact as described, but the screenshot at https://argoproj.github.io/argo-cd/developer-guide/ci/ does not match how CircleCI looks for me. I just have the Test Summary, Configuration, Timing, and Parameters tabs. No Artifact tab.

Comment (alexmt) [reply manifests/base/kustomization.yaml]: Ah, you are right. Circle CI is showing artifacts only to authenticated users. I could not find a way to enable it. Will change CI to just print the diff in logs.

**Review by alexmt — APPROVED** [review]
LGTM! @jutley , please sign [CLA](https://cla-assistant.io/argoproj/argo-cd?pullRequest=1889) and it is good to go.

**Comment by jutley** [comment]
I'll need to confirm with my org that I can sign the CLA. Should I clean up the commits? Lots of reverts cluttering up the history.

**Comment by alexmt** [comment]
Sure, there is no rush. No need to cleanup commit history, we squash PRs into one commit.

**Comment by jutley** [comment]
Please do not merge quite yet. I have signed the CLA, but this was by mistake. I logged in on the CLA signing page expecting for there to be another action to sign as that user, but it seems like logging in IS signing the CLA. This was by mistake. I need to get this resolved.

**Comment by jutley** [comment]
@alexmt Got approval for this contribution. Feel free to merge whenever you see fit!

**Comment by alexmt** [comment]
Awesome, thank you!

**Pull request merged**
